client/app/lib/components/sidebar/index.coffee

```diff
@@ -92,22 +93,35 @@ module.exports = class Sidebar extends React.Component
     MENU.destroy()
 
     draft = @state.drafts.get id
+    _id = draft.get '_id'
+
     switch title
-      when 'Edit' then router.handleRoute "/Stack-Editor/#{id}"
+      when 'Edit','ViewStack'
+        router.handleRoute "/Stack-Editor/#{id}"
       when 'Initialize'
         EnvironmentFlux.actions.generateStack(id).then ({ template }) ->
           appManager.tell 'Stackeditor', 'reloadEditor', template._id
       when 'Clone'
-        EnvironmentFlux.actions.cloneStackTemplate draft.toJS(), yes
+        remote.api.JStackTemplate.one { _id }, (err, template) =>
+          return @showErrorMessage 'Error occured while cloning template'  if err
+          EnvironmentFlux.actions.cloneStackTemplate template, (err) =>
+            return @showErrorMessage 'Error occured while cloning template'  if err
+            return @showErrorMessage 'Your stack template successfully cloned'
```

Comment:
it looks weird that we're showing a success message over `showErrorMessage` method.